src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt

```diff
@@ -369,13 +482,13 @@ class CreateMafVcf : CliktCommand(help = "Create gVCF and hVCF from Anchorwave M
         //create the asmHeader lines
         if(!asmHeaders.containsKey(assemblyHaplotypeHash)) {
             asmHeaders[assemblyHaplotypeHash] =
-                VCFAltHeaderLine(
-                    "<ID=${assemblyHaplotypeHash}, Description=\"haplotype data for line: ${metaDataRecord.sampleName}\">," +
-                            "Number=6,Source=\"${dbPath}/assemblies.agc\"," +
-                            "Contig=\"${metaDataRecord.asmContig}\",Start=\"${metaDataRecord.asmStart}\"," +
-                            "End=\"${metaDataRecord.asmEnd}\",Checksum=\"Md5\",RefRange=\"${refSeqHash}\">",
-                    VCFHeaderVersion.VCF4_2
-                )
+            VCFAltHeaderLine(
+                "<ID=${assemblyHaplotypeHash}, Description=\"haplotype data for line: ${metaDataRecord.sampleName}\">," +
+                        "Number=4,Source=\"${dbPath}/assemblies.agc\"," +
+                        "Regions=\"${metaDataRecord.asmRegions.map { "${it.first.contig}:${it.first.position}-${it.second.position}" }.joinToString(",")}\"," +
+                        "Checksum=\"Md5\",RefRange=\"${refSeqHash}\">",
```

Comment:
Is this a good idea to use a comma as a delimiter for the regions? Comma is the delimiter for fields in the ALT header.  This could have implications for parsing.  I'd like to see something else used as delimiter for regions.  You have colon and dash inside each region.  Would semi-colon work between regions?

Reply:
@zrm22 Actually, the comma is fine.   What we have wrong is that these fields should be quoted.  From the VCF 4.3 spec under section 1.4 Meta-information lines:

##INFO=<ID=ALLELEID,Number=A,Type=String,Description="Allele ID",Source="ClinVar",Version="20220804">
In the above example, the extra fields of “Source” and “Version” are provided. **The values of optional fields must be
written as quoted strings, even for numeric values.** Other structured lines not defined by this specification may also
be used; the only default field for such lines is the required ID field.

We need to put double quotes around all the fields we have added.

https://samtools.github.io/hts-specs/VCFv4.3.pdf

Reply:
These comments stemmed from looking at the created Ref.h.vcf which is now created and has lines that look like:
##ALT=<ID=06ae4e937668d301e325d43725a38c3f,Description="haplotype data for line: Ref",Number=4,Source="data/test/smallseq/Ref.fa",Regions=1:45001-49500,Checksum=Md5,RefRange=06ae4e937668d301e325d43725a38c3f>

Note that while the Description and Source field values are surrounded by double quotes, the Source, Regions, Checksum and RefRange fields are not.

However, in the code, the double quotes exist.  And I found when testing at scale, the assembly h.vcf files had double quotes around the Regions values when they were comprised of more than a single region.  See phg_v2 slack channel 11/20/23 for discussion.